### PR TITLE
feat(premium): subscription success screen

### PR DESCRIPTION
**Pull request by timc-mozilla**
## Summary
* Adds success screen for subscription purchase

## Implementation Details
* Added new image (setup for dark mode but currently dont have design) has a follow up ticket

## Test Steps
* complete purchase
* get taken to hooray screen

## PR Checklist:
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

## Screenshots
image

### CODE REVIEW

File: PocketKit/Sources/PocketKit/Premium/View/PremiumUpgradeSuccessView.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    var body: some View {
+        VStack(spacing: 20) {
+            dismissButton
+            Image(uiImage: UIImage(asset: .premiumHooray))
+                .resizable()
+                .scaledToFit()
+                .frame(width: 350, height: 350)
+            Text("Hooray!")
+                .style(.title)
+            Text("You’re officially a Pocket Premium member. Welcome to the new ad-free, customizable, permanent version of your Pocket. We think you’ll like it here.")
```
[the remark this answers is not in the document]
Comment (timc-mozilla) [reply PocketKit/Sources/PocketKit/Premium/View/PremiumUpgradeSuccessView.swift]: updated!

**Review by bassrock — APPROVED** [review]

File: PocketKit/Sources/PocketKit/Premium/View/PremiumUpgradeSuccessView.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private var dismissButton: some View {
+        HStack(spacing: 0) {
+            Spacer()
+            Button {
+                dismiss()
+            } label: {
+                Image(asset: .close).renderingMode(.template).foregroundColor(Color(.ui.grey5))
+            }
+            .padding(.top, 20)
+            .padding([.leading, .trailing], 20)
```
Comment (Gio2018) [inline PocketKit/Sources/PocketKit/Premium/View/PremiumUpgradeSuccessView.swift:47]: Can we group the magic numbers?

File: PocketKit/Sources/PocketKit/Settings/SettingsView.swift [hunk 3]
```diff
@@ -45,9 +45,30 @@ struct SettingsForm: View {
     var body: some View {
         Form {
             Group {
-                topSectionWithLeadingDivider()
```
Comment (Gio2018) [inline PocketKit/Sources/PocketKit/Settings/SettingsView.swift:48]: It's not clear to me why are we removing this now?

**Review by Gio2018 — CHANGES_REQUESTED** [review]
Hey @timc-mozilla , not sure if I'm missing something, but it looks like the code to handle the separator under "Go Premium" has been removed. The issue does not show in this branch because we don't have the image, but I don't think it should be removed, right?

Also, parts of the the Hooray screen image in dark mode are not clearly visible (see screenshot)
<p align=center>

</p>

File: PocketKit/Sources/PocketKit/Settings/SettingsView.swift (hunk 3 above)
Comment (timc-mozilla) [reply PocketKit/Sources/PocketKit/Settings/SettingsView.swift:48]: good catch. i missed it during the rebase. ill fix that in the next commit

File: PocketKit/Sources/PocketKit/Premium/View/PremiumUpgradeSuccessView.swift (hunk 2 above)
Comment (timc-mozilla) [reply PocketKit/Sources/PocketKit/Premium/View/PremiumUpgradeSuccessView.swift]: yup, will do.

**Comment by timc-mozilla** [comment]
too many merge conflicts. closing and cherry picking my changes

**Pull request closed without merging**